app/scripts/lib/transaction/metrics.ts

```diff
@@ -1146,3 +1165,67 @@ function allowanceAmountInRelationToTokenBalance(
   }
   return null;
 }
+
+async function addBatchProperties(
+  transactionMeta: TransactionMeta,
+  getMethodData: (data: string) => Promise<{ name?: string } | undefined>,
+  properties: Record<string, Json | undefined>,
+  sensitiveProperties: Record<string, Json | undefined>,
+) {
+  const isExternal = origin && origin !== ORIGIN_METAMASK;
+  const { delegationAddress, nestedTransactions, txParams } = transactionMeta;
+  const { authorizationList } = txParams;
+  const isBatch = Boolean(nestedTransactions?.length);
+  const isUpgrade = Boolean(authorizationList?.length);
+
+  if (isExternal) {
+    properties.api_method = isBatch
+      ? MESSAGE_TYPE.WALLET_SEND_CALLS
+      : MESSAGE_TYPE.ETH_SEND_TRANSACTION;
+  }
+
+  if (isBatch) {
+    properties.batch_transaction_count = nestedTransactions?.length;
+    properties.batch_transaction_method = 'eip7702';
+
+    properties.transaction_contract_method = await getNestedMethodNames(
+      nestedTransactions ?? [],
+      getMethodData,
+    );
+
+    sensitiveProperties.transaction_contract_address = nestedTransactions
+      ?.filter(
+        (tx) =>
+          tx.type === TransactionType.contractInteraction && tx.to?.length,
```

Comment:
why do we filter by contractInteraction transaction type here? Is it because in the future with different transaction types we don't want to leak "to" addresses that are not contracts into the metric properties?

Reply:
Exactly, this is specifically for contracts, so that type confirms we've checked code exists and it is a contract (or at least has code). This is the same as the existing logic, but I've just applied it for the array.